### PR TITLE
Shorten subdirectory IDs

### PR DESCRIPTION
**Pull request by nothingmuch**
WIP since this should probably be refactored a bit before merging (see TODO comments).

The message for the last commit contains a rationale for 64 bit values and hashing.

### CODE REVIEW

File: payjoin/Cargo.toml [hunk 1]
```diff
@@ -36,6 +36,7 @@ reqwest = { version = "0.12", default-features = false, optional = true }
 rustls = { version = "0.22.4", optional = true }
 url = "2.2.2"
 serde_json = "1.0.108"
+bitcoin_hashes = "0.15.0"
```
Comment (DanGould) [inline payjoin/Cargo.toml:39]: bitcoin 0.32 is using bitcoin_hashes 0.14. Unless we need a feature from 15, we ought not to duplicate the dependency

File: payjoin/src/lib.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@ pub use crate::receive::Error;
 pub mod send;
 
 #[cfg(feature = "v2")]
-pub(crate) mod hpke;
+pub mod hpke;
```
Comment (DanGould) [inline payjoin/src/lib.rs:31]: Why is this made public?

**Comment by DanGould** [comment]
Concept ACK in response to the "[Shorten subdirectory IDs to 64 pseudorandom bits](https://github.com/payjoin/rust-payjoin/pull/386/commits/a6b64eda4d19983043e92641d87cbe1c1d9953e3)" commit message.

✅ utxo set size ≈ 2^28
✅ daily tx count limit ≈ 2^21

> the probability for an individual session to
experience a random collision is << 1e-10

10 9s is way more than enough reliability of this particular component, and the utxo set size and even daily tx counts are already way overestimations of the number of sessions a single directory will ever have and many directories may be run.


Concept ACK also on your note that this already improves collision resistance on the naive truncated shortIDs of not-completely-pseudorandom values.

File: payjoin/src/lib.rs [hunk 3]
```diff
@@ -30,6 +30,8 @@ pub mod send;
 #[cfg(feature = "v2")]
 pub(crate) mod hpke;
 #[cfg(feature = "v2")]
+pub use hpke::{HpkeKeyPair, HpkePublicKey};
```
Comment (DanGould) [inline payjoin/src/lib.rs:33]: should be 
`pub use crate::hpke::{HpkeKeyPair, HpkePublicKey};`
to be compatible with msrv

**Review by DanGould — APPROVED** [review]
This appears to do what it says it does to me. There are quite a few TODOs that are OK to do in follow ups. I find it a bit weird to change the API in the first commit and only to use it in later commits but that's a nit of nits

File: payjoin-directory/src/db.rs [hunk 4]
```diff
@@ -99,6 +117,6 @@ impl DbPool {
     }
 }
 
-fn channel_name(pubkey_id: &str, channel_type: &str) -> String {
-    format!("{}:{}", pubkey_id, channel_type)
+fn channel_name(pubkey_id: &ShortId, channel_type: &str) -> Vec<u8> {
```
Comment (spacebear21) [inline payjoin-directory/src/db.rs:120]: This helper is no longer that helpful IMO and could be replaced by direct calls to `column_key()` in later commits.

**Review by spacebear21 — APPROVED** [review]
utACK a95a2ba00c324cd669ad525400f227c556726c2b. Noted one more nit that can be addressed in a follow-up.

**Pull request merged**